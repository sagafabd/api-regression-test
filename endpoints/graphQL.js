import supertest from 'supertest';
import { jsonToGraphQLQuery } from 'json-to-graphql-query';
import dotenv from 'dotenv';

dotenv.config();

const api = supertest(process.env.BASE_URL)

export default (token,body) => api.post('/graphql')
//.set("Authorization", token)
.send(
    {
        query : jsonToGraphQLQuery(body, { pretty: true })
    }
)