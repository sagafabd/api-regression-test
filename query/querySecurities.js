export const getEvent = () => ({
    query: {
        events: {
                _id: true,
                title: true,
                description: true,
                price: true,
                date: true
        }
    }
})

export const seedInsertEvent = (eventInput) => ({
    mutation: {
        createEvent: {  
            __args: eventInput,
            _id: true,
                title: true,
                description: true,
                price: true,
                date: true
            }  
    }
})

