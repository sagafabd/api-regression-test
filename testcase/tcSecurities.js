export const eventTestCase={
    positiveCase: {
        describe: "Get Securities - Positive Case",
        successResult: "Get all data securities and success response",
        validateDataById: "Get data by id securities",
        validateDataByName: "Get data by name securities",
        createNewData: "Create data securities",
        updateNewData: "Update data securities",
        deleteDataById: "Delete data by id securities",
        compareData: "Compare data result securities",
    },
    negativeCase: {
        describe: "Get Securities - Negative Case",
        invalidToken: "Get all data securities and invlid token",
        blankToken: "Get all data securities and blank token",
        errorCode500: "Get all data securities and error code 500",
        blankToken: "Get all data securities and blank token",
    }
}