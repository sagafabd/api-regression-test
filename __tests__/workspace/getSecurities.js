import endpoints from '../../endpoints/graphQL';
import * as query from '../../query/querySecurities';
import * as tc from '../../testcase/tcSecurities';
import dotenv from 'dotenv';

const token = process.env.TOKEN

describe(tc.eventTestCase.positiveCase.describe, () => {   

    // Create New Data
    // beforeAll(async () => {    
    //     const paramI = {
    //         eventInput: ( {
    //             title: "D Test", 
    //             description: "is works", 
    //           	price: 0.90,
    //             date: "2010-10-29T19:16:11.788Z"
    //             } 
    //         )
    //     }        
    //     const response = await endpoints(token, query.seedInsertEvent(paramI));
    //     expect(response.status).toEqual(200);
    //     expect(JSON.parse(response.text)).toMatchSnapshot();
    // });    
    
    test(tc.eventTestCase.positiveCase.successResult, async () => {
        const paramQ = {
            first: 10          
        }
        
        const response = await endpoints(token, query.getEvent());
        expect(response.status).toEqual(200);
        expect(JSON.parse(response.text)).toMatchSnapshot();
    })

})

