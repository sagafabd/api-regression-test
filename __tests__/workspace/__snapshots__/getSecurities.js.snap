// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Securities - Positive Case Get all data securities and success response 1`] = `
Object {
  "data": Object {
    "events": Array [
      Object {
        "_id": "61f864e643e1310fdcef2162",
        "date": "1572376571788",
        "description": "this works",
        "price": 2.9,
        "title": "F Test",
      },
      Object {
        "_id": "61f865fb93fdc7f8b9a76b0d",
        "date": "1572376571788",
        "description": "this works",
        "price": 2.9,
        "title": "F Test",
      },
      Object {
        "_id": "61f866ffa6923e9c769375cb",
        "date": "1572376571788",
        "description": "this works",
        "price": 2.9,
        "title": "F Test",
      },
      Object {
        "_id": "61f8674135881e7ef4ea3edd",
        "date": "1572376571788",
        "description": "this works",
        "price": 2.9,
        "title": "G Test",
      },
      Object {
        "_id": "61f8677cab5dbe20cc21aa47",
        "date": "1572376571788",
        "description": "this works",
        "price": 2.9,
        "title": "H Test",
      },
      Object {
        "_id": "61f87181ab5dbe20cc21aa4c",
        "date": "1288379771788",
        "description": "is works",
        "price": 0.9,
        "title": "D Test",
      },
      Object {
        "_id": "61f87192ab5dbe20cc21aa51",
        "date": "1288379771788",
        "description": "is works",
        "price": 0.9,
        "title": "D Test",
      },
    ],
  },
}
`;
